engines/scalac/src/main/scala/scala/macros/internal/engines/scalac/trees/Abstracts.scala

```diff
@@ -181,12 +200,19 @@ trait Abstracts extends scala.macros.trees.Abstracts with Positions { self: Univ
 
     object TermApply extends TermApplyCompanion {
       def apply(fun: Term, args: List[Term]): Term = g.Apply(fun, args)
-      def unapply(gtree: Any): Option[(Term, List[Term])] = ???
+      def unapply(gtree: Any): Option[(Term, List[Term])] = gtree match {
+        case gtree @ g.Apply(fun, args) if LitSymbol.symApply != gtree.symbol =>
```

Comment:
I don't understand the guard `if LitSymbol.symApply != gtree.symbol `. I guess it's something scalac-specific? Just curious.

Reply:
This was to support the test `syntax('S) == "'S"`, scalac desugars it into `scala.Symbol.apply("S")`. I've added a note saying we may want to skip this resugaring later down the road, this was just an experiment to support symmetric structure for literals.